sdk/core/azure-core/tests/test_basic_transport.py

```diff
@@ -0,0 +1,380 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from six.moves.http_client import HTTPConnection
+import time
+
+from azure.core.pipeline.transport import HttpRequest, HttpResponse, RequestsTransport
+from azure.core.pipeline.transport.base import HttpClientTransportResponse, _deserialize_response, MultiPartHelper
+from azure.core.pipeline.policies import HeadersPolicy
+from azure.core.pipeline import Pipeline
+
+
+def test_http_request_serialization():
+    # Method + Url
+    request = HttpRequest("DELETE", "/container0/blob0")
+    serialized = request.serialize()
+
+    expected = (
+        b'DELETE /container0/blob0 HTTP/1.1\r\n'
+        # No headers
+        b'\r\n'
+    )
+    assert serialized == expected
+
+    # Method + Url + Headers
+    request = HttpRequest(
+        "DELETE",
+        "/container0/blob0",
+        headers={
+            "x-ms-date": "Thu, 14 Jun 2018 16:46:54 GMT",
+            "Authorization": "SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=",
+            "Content-Length": "0",
+        }
+    )
+    serialized = request.serialize()
+
+    expected = (
+        b'DELETE /container0/blob0 HTTP/1.1\r\n'
+        b'x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT\r\n'
+        b'Authorization: SharedKey account:G4jjBXA7LI/RnWKIOQ8i9xH4p76pAQ+4Fs4R1VxasaE=\r\n'
+        b'Content-Length: 0\r\n'
+        b'\r\n'
+    )
+    assert serialized == expected
+
+
+    # Method + Url + Headers + Body
+    request = HttpRequest(
+        "DELETE",
+        "/container0/blob0",
+        headers={
+            "x-ms-date": "Thu, 14 Jun 2018 16:46:54 GMT",
+        },
+    )
+    request.set_bytes_body(b"I am groot")
+    serialized = request.serialize()
+
+    expected = (
+        b'DELETE /container0/blob0 HTTP/1.1\r\n'
+        b'x-ms-date: Thu, 14 Jun 2018 16:46:54 GMT\r\n'
+        b'Content-Length: 10\r\n'
+        b'\r\n'
+        b'I am groot'
+    )
+    assert serialized == expected
+
+
+def test_http_client_response():
+    # Create a core request
+    request = HttpRequest("GET", "www.httpbin.org")
+
+    # Fake a transport based on http.client
+    conn = HTTPConnection("www.httpbin.org")
+    conn.request("GET", "/get")
+    r1 = conn.getresponse()
+
+    response = HttpClientTransportResponse(request, r1)
+
+    # Don't assume too much in those assert, since we reach a real server
+    assert response.internal_response is r1
+    assert response.reason is not None
+    assert response.status_code == 200
+    assert len(response.headers.keys()) != 0
+    assert len(response.text()) != 0
+    assert "content-type" in response.headers
```

Comment:
Are we intended to check content-type in response.headers twice?

Reply:
Yes, to verify the headers are case-insensitive